modin/core/dataframe/pandas/metadata/dtypes.py

```diff
@@ -12,13 +12,518 @@
+            elif dtype is None and self._remaining_dtype is not None:
+                result[col] = self._remaining_dtype
+            else:
+                result[col] = dtype
+        remaining_dtype = self._remaining_dtype if len(unknown_cols) != 0 else None
+        return DtypesDescriptor(
+            result,
+            unknown_cols,
+            remaining_dtype,
+            self._parent_df,
+            columns_order=columns_order,
+        )
+
+    def copy(self) -> "DtypesDescriptor":
+        """
+        Get a copy of this descriptor.
+
+        Returns
+        -------
+        DtypesDescriptor
+        """
+        return type(self)(
+            self._known_dtypes.copy(),
+            self._cols_with_unknown_dtypes.copy(),
+            self._remaining_dtype,
+            self._parent_df,
+            columns_order=None
+            if self.columns_order is None
+            else self.columns_order.copy(),
+            know_all_names=self._know_all_names,
+            _schema_is_known=self._schema_is_known,
+        )
+
+    def set_index(
+        self, new_index: Union[pandas.Index, "ModinIndex"]
+    ) -> "DtypesDescriptor":
+        """
+        Set new column names for this descriptor.
+
+        Parameters
+        ----------
+        new_index : pandas.Index or ModinIndex
+
+        Returns
+        -------
+        DtypesDescriptor
+            New descriptor with updated column names.
+
+        Notes
+        -----
+        Calling this method on a descriptor that returns ``None`` for ``.columns_order``
+        will result into information lose.
+        """
+        if self.columns_order is None:
+            # we can't map new columns to old columns and lost all dtypes :(
+            return DtypesDescriptor(
+                cols_with_unknown_dtypes=new_index,
+                columns_order={i: col for i, col in enumerate(new_index)},
+                parent_df=self._parent_df,
+                know_all_names=True,
+            )
+
+        new_self = self.copy()
+        renamer = {old_c: new_index[i] for i, old_c in new_self.columns_order.items()}
+        new_self._known_dtypes = {
+            renamer[old_col]: value for old_col, value in new_self._known_dtypes.items()
+        }
+        new_self._cols_with_unknown_dtypes = [
+            renamer[old_col] for old_col in new_self._cols_with_unknown_dtypes
+        ]
+        new_self._columns_order = {
+            i: renamer[old_col] for i, old_col in new_self._columns_order.items()
+        }
+        return new_self
+
+    def equals(self, other: "DtypesDescriptor") -> bool:
+        """
+        Compare two descriptors for equality.
+
+        Parameters
+        ----------
+        other : DtypesDescriptor
+
+        Returns
+        -------
+        bool
+        """
+        return (
+            self._known_dtypes == other._known_dtypes
+            and set(self._cols_with_unknown_dtypes)
+            == set(other._cols_with_unknown_dtypes)
```

Comment:
Why are you using `set` here?

Reply:
just to ignore the order

Reply:
We can ignore it here because we there is the following check: `self.columns_order == other.columns_order`?

Reply:
nope, consider this example:
```python
dt1 = DtypesDescriptor(cols_with_unknown_dtypes=["a", "b"], columns_order={0: "a", 1: "b"})
dt2 = DtypesDescriptor(cols_with_unknown_dtypes=["b", "a"], columns_order={0: "a", 1: "b"})
dt1.equals(dt2) # should be true
```

Reply:
I see, thanks